### PR TITLE
[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa

### PR DESCRIPTION
**Pull request by larryliu0820**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #4687

Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

### CODE REVIEW

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

File: examples/models/llava/install_requirements.sh [hunk 1]
```diff
@@ -6,39 +6,7 @@
...
-# The deps of llava can have different versions than deps of ExecuTorch.
-# For example, torch version required from llava is older than ExecuTorch.
-# To make both work, recover ExecuTorch's original dependencies by rerunning
-# the install_requirements.sh. Notice this won't install executorch.
-bash -x ./install_requirements.sh --pybind xnnpack
-
-# Newer transformer (4.38) will give TypeError: LlavaLlamaForCausalLM.forward() got an unexpected keyword argument 'cache_position'
-pip install timm==0.6.13
-pip install transformers==4.37.2
+pip install transformers
```
Comment (mcr229) [inline examples/models/llava/install_requirements.sh:10]: Should we also run 
`bash ./install_requirements` and
`bash examples/models/llama2/install_requirements.sh` as well

File: .github/workflows/pull.yml [hunk 2]
```diff
@@ -204,28 +204,14 @@ jobs:
         conda activate "${CONDA_ENV}"
 
         PYTHON_EXECUTABLE=python bash .ci/scripts/setup-linux.sh "cmake"
+        
+        # install pybind
+        bash install_requirements.sh --pybind xnnpack
 
         # install Llava requirements
         bash examples/models/llama2/install_requirements.sh
         bash examples/models/llava/install_requirements.sh
```
Comment (mcr229) [inline .github/workflows/pull.yml:214]: why don't we run export_llava.py anymore? looks like test_llava still requires the model to have already been exported?

**Review by mcr229 — APPROVED** [review]

File: examples/models/llava/install_requirements.sh (hunk 1 above)
Comment (larryliu0820) [reply examples/models/llava/install_requirements.sh:10]: I think user should call it outside

File: .github/workflows/pull.yml (hunk 2 above)
Comment (larryliu0820) [reply .github/workflows/pull.yml:214]: The last test in test_llava.py exports and tests llava

File: examples/models/llava/install_requirements.sh (hunk 1 above)
Comment (mcr229) [reply examples/models/llava/install_requirements.sh:10]: I see, the ./install_requirements makes sense, but at least for the llama2, I don't think its immediately obvious that you have to install the requirements for llama2 in order for llava to work.

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

File: examples/models/llava/install_requirements.sh (hunk 1 above)
Comment (guangy10) [reply examples/models/llava/install_requirements.sh:10]: @larryliu0820 great to me we are moving to the HF model. I think we should have no blocker to clean up all the ad-hoc setup requirements https://github.com/pytorch/executorch/pull/4320?

**Pull request merged**